### PR TITLE
Fix time issue and Toast not showing when notification are disabled

### PR DESCRIPTION
**Pull request by lgoix**

### CODE REVIEW

File: app/src/main/java/com/geeksville/mesh/service/MeshService.kt [hunk 1]
```diff
@@ -942,22 +943,22 @@ class MeshService : Service(), Logging {
         val p = packet.decoded
 
         // If the rxTime was not set by the device (because device software was old), guess at a time
-        val rxTime = if (packet.rxTime == 0) packet.rxTime else currentSecond()
+        val rxTime = if (packet.rxTime != 0) packet.rxTime else currentSecond()
```
Comment (lgoix) [inline app/src/main/java/com/geeksville/mesh/service/MeshService.kt:946]: @geeksville : I think this was not good if rxTime was 0 if not present in the packet

Comment (geeksville) [reply app/src/main/java/com/geeksville/mesh/service/MeshService.kt:946]: @lgoix hmm - I don't quite understand.  if rxTime is not specified by the device, why would we not want to pick the local phone time?

File: app/src/main/java/com/geeksville/mesh/MainActivity.kt [hunk 2]
```diff
@@ -869,7 +869,7 @@ class MainActivity : AppCompatActivity(), Logging,
         try {
             val packageInfo: PackageInfo = packageManager.getPackageInfo(packageName, 0)
             val versionName = packageInfo.versionName
-            Toast.makeText(this, versionName, Toast.LENGTH_LONG).show()
+            Toast.makeText(applicationContext, versionName, Toast.LENGTH_LONG).show()
```
Comment (geeksville) [inline app/src/main/java/com/geeksville/mesh/MainActivity.kt:872]: smart!

File: app/src/main/java/com/geeksville/mesh/service/MeshService.kt (hunk 1 above)
Comment (lgoix) [reply app/src/main/java/com/geeksville/mesh/service/MeshService.kt:946]: Yes that is the fix

It was: if rxtime is set then use the alocal phone time

File: app/src/main/java/com/geeksville/mesh/MainActivity.kt (hunk 2 above)
Comment (lgoix) [reply app/src/main/java/com/geeksville/mesh/MainActivity.kt:872]: rxTime=0 when packet.rxTime==0

Comment (lgoix) [reply app/src/main/java/com/geeksville/mesh/MainActivity.kt:872]: The code was:
```
val rxTime = if (packet.rxTime == 0) packet.rxTime else currentSecond()
```

If device has rxTime, we were using currentSecond()

```
if (packet.rxTime == 0)
   val rxTime = 0
else
   val rxTime = currentSecond()
```

Comment (geeksville) [reply app/src/main/java/com/geeksville/mesh/MainActivity.kt:872]: @lgoix oh I see!  I'm a moron and was not thinking carefully!  Thank you for fixing this.  Sorry for being silly!

**Pull request merged**